pkg/auth/token_provider.go

```diff
@@ -19,33 +19,43 @@ package auth
 import (
 	"context"
 	"fmt"
+	"time"
 
 	"go.uber.org/zap"
 	authv1 "k8s.io/api/authentication/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/cache"
 	"k8s.io/client-go/kubernetes"
 	kubeclient "knative.dev/pkg/client/injection/kube/client"
 	"knative.dev/pkg/logging"
 )
 
+const (
+	expirationBufferTime = time.Second * 30
+)
```

Comment:
I'd make this higher to also account for clock skew which could be in the minutes order at times, something like 5 minutes seems safer

Reply:
@Cali0707 @creydr any thoughts here?

Reply:
clock skew is a good point. I'll create an issue to fix it

Reply:
created #7351

Reply:
Oh whoops, sorry I missed this! @pierDipi or @creydr - what is clock skew in this context? I've only encountered it before in digital circuits